Finished MarketPlace Screen

amjiao opened the pull request:
The Marketplace screen for now.

I don't think I have a full list of all the filters so I'll add those later. 

![image](https://github.com/user-attachments/assets/ef85a9cc-ccda-4107-918a-6afb44a488db)

amjiao committed:
Finished MarketPlace Screen

amjiao committed:
Merge branch 'main' into Amy/marketplace-screen

AndrewCheung360 commented on app/src/main/java/com/appdev/all_in_android/ui/screens/MarketPlaceScreen.kt:
```diff
@@ -1,12 +1,132 @@
+                FilterRow()
+
+                LazyColumn{
+                    items(count = 5){
```
Some small changes, but I think it would be better if we pass in a list as a parameter to the marketplace screen composable instead of defining it directly here. I defined a Contract data class under the models package, which you can maybe use as a type for the List.

AndrewCheung360 commented on app/src/main/java/com/appdev/all_in_android/ui/screens/MarketPlaceScreen.kt:
```diff
@@ -1,12 +1,132 @@
+        contentPadding = PaddingValues(0.dp),
+        modifier = Modifier.padding(2.dp)
+    ){
+        Text("pts")
```
I think we should have the FilterButton take in a filterName parameter instead of defining pts

AndrewCheung360 commented on app/src/main/java/com/appdev/all_in_android/ui/screens/MarketPlaceScreen.kt:
```diff
@@ -1,12 +1,132 @@
+}
+
+@Composable
+fun FilterRow(){
```
I think it would be better to pass in a list of filters as a parameter for FilterRow.

AndrewCheung360 reviewed (commented):
Hey Amy, the screen looks great! I just made a couple of comments about converting some of the hard-coded data to parameters to pass in to the  composables instead for future usages of the market place screen. Other than that, you did a great job with all of the various marketplace components!

oliviaxjiang commented:
the PR looks great! I agree with Andrew that FilterButton can take in some parameters so it is less hard-coded. Upon looking at how Eatery does it, I think it would make sense for the FilterButton here to take in a onClick and text to display. As for FilterRow, I think we can either change it so it takes in a list of filters or we can wait until design is clearer with what these Filters do. If there is only one FilterRow in the app or if the FilterRow always have the same content, I don't think its necessary to have the list of filters as a parameter. But anyways thank u for the great work and no rush in getting this done! Filter logic can be complicated and we can talk about it together during the next work session as well.

amjiao committed:
Added dummy data for player cards

amjiao committed:
Parameterized Marketplace screen

amjiao committed:
Updated MarketplaceScreen call in NavigationStack.kt

amjiao committed:
Miscellaneous gradle updates

AndrewCheung360 commented on app/src/main/java/com/appdev/all_in_android/ui/screens/MarketPlaceScreen.kt:
```diff
@@ -1,12 +1,179 @@
+
+        Column(
+            modifier = Modifier
+                .padding(padding)
```
![image](https://github.com/user-attachments/assets/bd8a08ff-ebe3-4cb1-8f14-7443b5ffa32c)
I think the bottom padding for both columns might be causing this to screen to be cut a little short with the navbar

AndrewCheung360 replied:
Also, I don't believe this is a problem with the marketplacescreen implementation necessarily, but do the gain and action type show up for you when you add this to the navigation?

AndrewCheung360 commented on app/src/main/java/com/appdev/all_in_android/ui/screens/MarketPlaceScreen.kt:
```diff
@@ -1,12 +1,179 @@
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun MarketplaceScreen(){
```
Very minor, but would it be possible to either change the composable name or the file name such that they match each other exactly?

amjiao replied:
done!

AndrewCheung360 commented on app/src/main/java/com/appdev/all_in_android/ui/screens/MarketPlaceScreen.kt:
```diff
@@ -1,12 +1,179 @@
+                )
+                FilterRow()
+
+                LazyColumn{
```
I think you can accomplish something similar with the 2 cards side by side with a LazyVerticalGrid, which automatically divides the items of a list into two columns if you set it in the properties. This is completely up to you of course, as your current implementation works and I don't see any issue with it.

amjiao replied:
Decided to go with LazyVerticalGrid, it's a lot cleaner, thanks!

amjiao committed:
Updated MarketplaceScreen.kt

- Style fixes on code parameterization, filter row, and usage of LazyVerticalGrids.
- Fixed padding issues

amjiao committed:
Rename MarketPlaceScreen.kt to MarketplaceScreen.kt

oliviaxjiang reviewed (approved):
looks good!
